filters/include/pcl/filters/experimental/transform_filter.h

```diff
@@ -0,0 +1,87 @@
+/*
+ * SPDX-License-Identifier: BSD-3-Clause
+ *
+ *  Point Cloud Library (PCL) - www.pointclouds.org
+ *  Copyright (c) 2020-, Open Perception
+ *
+ *  All rights reserved
+ */
+
+#pragma once
+
+#include <pcl/common/point_tests.h>
+#include <pcl/filters/filter.h>
+
+namespace pcl {
+namespace experimental {
+
+/**
+ * \brief The base of GridFilterBase which has the common member functions and
+ * attribute for the grid filters.
+ * \ingroup filters
+ */
+template <typename GridStruct, typename PointT>
+class TransformFilter : public Filter<PointT>, public GridStruct {
+protected:
+  using Filter<PointT>::filter_name_;
+  using Filter<PointT>::getClassName;
+  using Filter<PointT>::input_;
+
+  using PointCloud = typename Filter<PointT>::PointCloud;
+  using PointCloudPtr = typename PointCloud::Ptr;
+  using PointCloudConstPtr = typename PointCloud::ConstPtr;
+
+public:
+  /** \brief Empty constructor. */
+  TransformFilter()
```

Comment:
Removing the default ctor doesn't negate the fact that GridStruct needs to be default constructible.

Reply:
We agreed on having `GridStruct` as an object in `TransformFilter`